frontend/src/lib/components/ui/heatmap/heatmap.svelte

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+  import {
+    HeatmapChart,
+    ScaleTypes,
+    type AreaChartOptions,
+    type HeatmapChartOptions,
+  } from '@carbon/charts-svelte';
+  import '@carbon/charts-svelte/styles.css';
+
+  export let data: Array<{
+    dayOfWeek: number;
+    hourOfDay: number;
+    count: number;
+  }>;
+
+  const hourOfDayFormatter = new Intl.DateTimeFormat(undefined, {
+    hour: 'numeric',
+    hour12: false,
+  });
+  const dayOfWeekFormatter = new Intl.DateTimeFormat(undefined, {
+    weekday: 'long',
+  });
+
+  const dayOfWeekLabels = Array.from(
+    { length: 7 },
+    (_, i) => {
+      const date = new Date();
+      date.setDate(date.getDate() - date.getDay() + i + 1); // we + 1 to make 0 = monday
+      return dayOfWeekFormatter.format(date);
+    }
+    // dayOfWeekFormatter.format(new Date(2024, 8, i)) // Month that starts on a Monday (could be any - just for sorting)
+  );
+  const hourOfDayLabels = Array.from({ length: 24 }, (_, i) =>
+    hourOfDayFormatter.format(new Date(0, 0, 0, i))
+  );
+
+  console.table(data);
+  console.log(dayOfWeekLabels);
+
+  const mappedChartData =
+    data?.map((stat) => ({
+      day: dayOfWeekLabels[stat.dayOfWeek + 1], // we + 1 to make 0 = monday
```

Comment:
I'm not sure I understand this `+ 1 to make 0 = monday`. Don't we have to `% 7`?

Reply:
Same goes for above.

Reply:
The values 0-6 is defined by postgresql extract function: https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT

They define sunday as 0.

But with that being said, this logic was way off and did not show the correct data - but I have fixed it now